constraints/miden-vm/stack.air

```diff
@@ -396,320 +493,396 @@ ev is_unchanged(main: [values[4]]):
+        or([stack]) when op_flags[37]
+        u32and([stack], [b_chip]) when op_flags[38]
+        u32xor([stack], [b_chip]) when op_flags[39]
+        # TODO: add match variant for caller 
+        frie2f4()
+        drop([stack]) when op_flags[41]
+        cswap([stack]) when op_flags[42]
+        cswapw([stack]) when op_flags[43]
+        mloadw([stack], [clk, b_chip]) when op_flags[44]
+        mstore([stack], [helper[3], clk, b_chip]) when op_flags[45]
+        mstorew([stack], [clk, b_chip]) when op_flags[46]
+        fmpupdate([stack], [fmp]) when op_flags[47]
+
+        pad([stack]) when op_flags[48]
+        dup([stack]) when op_flags[49]
+        dup1([stack]) when op_flags[50]
+        dup2([stack]) when op_flags[51]
+        dup3([stack]) when op_flags[52]
+        dup4([stack]) when op_flags[53]
+        dup5([stack]) when op_flags[54]
+        dup6([stack]) when op_flags[55]
+        dup7([stack]) when op_flags[56]
+        dup9([stack]) when op_flags[57]
+        dup11([stack]) when op_flags[58]
+        dup13([stack]) when op_flags[59]
+        dup15([stack]) when op_flags[60]
+        advpop([stack]) when op_flags[61]
+        sdepth([stack, bookkeeping]) when op_flags[62]
+        clk([stack], [clk]) when op_flags[63]
+
+
 
 # No stack shift operations
 
-ev noop(main: [s[16]], aux: [b[7]]):
-    enf f_dup(b) * delta(elem) = 0 for elem in stack
+ev noop(main: [s[16]]):
+    enf delta(elem) = 0 for elem in s
 
-ev eqz(main: [s[16], helper], aux: [b[7]]):
-    enf f_eqz(b) * (s[0]' * s[0]) = 0
-    enf f_eqz(b) * (s[0]' - (1 - s[0] * helper))
-    enf f_eqz(b) * delta(s[i]) = 0 for i in 1..16
+ev eqz(main: [s[16], h0]):
+    enf s[0]' * s[0] = 0
+    enf s[0]' = 1 - s[0] * h0
+    enf delta(s[i]) = 0 for i in 1..16
 
-ev neg(main: [s[16]], aux: [b[7]]):
-    enf f_neg(b) * (s[0]' + s[0]) = 0
-    enf f_neg(b) * delta(s[i]) = 0 for i in 1..16
+ev neg(main: [s[16]]):
+    enf s[0]' + s[0] = 0
+    enf delta(s[i]) = 0 for i in 1..16
 
-ev inv(main: [s[16]], aux: [b[7]]):
-    enf f_inv(b) * (1 - s[0]' * s[0]) = 0
-    enf f_inv(b) * delta(s[i]) = 0 for i in 1..16
+ev inv(main: [s[16]]):
+    enf s[0]' * s[0] = 1
+    enf delta(s[i]) = 0 for i in 1..16
 
-ev incr(main: [s[16]], aux: [b[7]]):
-    enf f_incr(b) * (s[0]' - s[0] - 1) = 0
-    enf f_incr(b) * delta(s[i]) = 0 for i in 1..16
+ev incr(main: [s[16]]):
+    enf s[0]' = s[0] + 1
+    enf delta(s[i]) = 0 for i in 1..16
 
-ev not(main: [s[16]], aux: [b[7]]):
-    enf f_not(b) * binary_check_sub(s[0])
-    enf f_not(b) * (s[0]' - !s[0]) = 0
-    enf f_not(b) * delta(s[i]) = 0 for i in 1..16
+ev not(main: [s[16]]):
+    enf is_binary([s[0]])
+    enf s[0]' = !s[0]
+    enf delta(s[i]) = 0 for i in 1..16
 
-ev fmpadd(main: [s[16]], aux: [b[7], fmp]):
-    enf f_fmpadd(b) * (s[0]' - s[0] - fmp) = 0
-    enf f_fmpadd(b) * delta(s[i]) = 0 for i in 1..16
+ev fmpadd(main: [s[16]], aux: [fmp]):
+    enf s[0]' = s[0] + fmp
+    enf delta(s[i]) = 0 for i in 1..16
```

Comment:
`fmp` is a part of main trace (not auxiliary trace). So, this should look like this:
```
ev op_fmpadd(main: [s[16], fmp]):
  enf s[0]' = s[0] + fmp
  enf s[i]' = s[i] for i in 1..16
```

Reply:
Ah, indeed, my bad. And I also made the same mistake with saying `v'` in the decoder PR calling `..._next` flags.